src/gen/async_generate_data_gemini.py

```diff
@@ -0,0 +1,99 @@
+import asyncio
+import aiohttp
+from typing import List, Dict
+import os
+import sys
+import json
+import numpy as np
+from datasets import load_dataset
+
+
+if os.getenv("COLAB_RELEASE_TAG"):
+    from google.colab import userdata
+
+    GOOGLE_API_KEY = userdata.get("GEMINI_ONE_API_KEY", None)
+else:
+    GOOGLE_API_KEY = os.getenv("GEMINI_ONE_API_KEY", None)
+
+assert GOOGLE_API_KEY, "GOOGLE_API_KEY cannot be None."
+
+
+MODEL_NAME = "gemini-1.0-ultra-latest"
+NUM_DATASET_SAMPLES = 5
+EXISTING_DATASET = load_dataset("sayakpaul/no_robots_only_coding", split="train_sft")
+RATE_LIMIT_PER_MINUTE = 60
+REQUEST_INTERVAL = 60 / RATE_LIMIT_PER_MINUTE  # Seconds between requests
+
+
+# prompt template can be loaded from a `.toml` file.
+def craft_prompt(instruction, response):
+    prompt = """
+Generate a series of (instruction, response) pairs that are similar in context and structure to the example provided below. Each pair should consist of a concise instruction followed by an appropriate, detailed response. The instruction should pose a clear task or question, while the response should provide a comprehensive answer or solution that could be understood by someone with a basic understanding of the subject.
+
+Example pair:
+
+Instruction: {instruction}
+Response: {response}
+
+Your task is to generate more pairs that maintain this level of clarity and detail. The topic is Coding. Ensure that the responses are informative and accurate, suitable for an educational context.
+
+Store the generated pairs in JSON format, with each pair as an object within an array. Each object should have two key-value pairs: "instruction" and "response". For instance:
+
+[{{"instruction": "text", "response": "text"}}, {{"instruction": "text", "response": "text"}}, ...]
+
+Remember to maintain consistency in the format and ensure the generated pairs are diverse and cover a broad range of subjects. You must return the response
+in the asked format and you must not add any additional text in your response.
+"""
+    return prompt.format(instruction=instruction, response=response)
+
+
+async def generate_text(prompt):
+    url = (
```

Comment:
Why manually creating requests without using the gemini library?

Reply:
`asyncio` works best with REST. Not with the wrapper APIs.

Reply:
wrapper is there for abstraction like we don't need to specify the hard-coded endpoint or something. If some underlying information from version to version changes, our code should be changed too.

But, do you still think calling REST natively is better. in terms of performance? Do you feel any significant gaps in between? I am just curious.

Reply:
The endpoint won't likely change either. We just have to supply the model name (and API key) and be done with it. 

Rest of the parameters need to be crafted for the wrapper API too. 

It eliminates a dependency too (`generativeai`). So that is another factor.